chameleon-server/src/main/java/com/chameleonvision/vision/pipeline/pipes/SolvePNPPipe.java

```diff
@@ -286,118 +308,69 @@ private MatOfPoint2f findBoundingBoxCorners(StandardCVPipeline.TrackedTarget tar
 
-        var boundingTl = new Point(
-            xList.get(0), yList.get(0)
-        );
-        var boundingBr = new Point (
-            xList.get(2), yList.get(2)
-        );
-        System.out.println("tl/br:\n" + boundingTl.toString() + "\n" + boundingBr.toString());
+    private boolean shouldRefineCorners = true;
 
-        var slightlyBiggerTl = new Point(
-            Math.max(0, boundingTl.x - 5),
-            Math.max(0, boundingTl.y - 5)
-        );
-
-        var slightlyBiggerBr = new Point(
-            Math.min(srcImage.rows(), boundingBr.x + 5),
-            Math.min(srcImage.cols(), boundingBr.y + 5)
-        );
-        var rect = new Rect(slightlyBiggerTl, slightlyBiggerBr);
-
-        var croppedImage = srcImage.submat(rect);
-        var corners = new MatOfPoint();
-        Imgproc.goodFeaturesToTrack(croppedImage, corners, 0,0.01,5);
-
-        List<Point> cornerList = new ArrayList<>(corners.toList());
-//        if(cornerList.size() != 8 && cornerList.size() != 4) return null;
-        cornerList.sort(leftRightComparator);
-
-        cornerList = cornerList.stream().map(point ->
-            new Point(point.x + slightlyBiggerTl.x, point.y + slightlyBiggerTl.y))
-            .collect(Collectors.toList());
-
-        // of these, we want the two leftmost and two rightmost points
-        var left1 = cornerList.get(0);
-        var left2 = cornerList.get(1);
-        var right1 = cornerList.get(0);
-        var right2 = cornerList.get(1);
-
-        // TODO maximize distance from the center rather than naively assume the leftmost and rightmost
-        // will have to do per quadrant
-
-        var leftOrder = left1.y < left2.y;
-        var rightOrder = right1.y < right2.y;
-
-        var tl = leftOrder ? left1 : left2;
-        var bl = !leftOrder ? left1 : left2;
-        var tr = rightOrder ? right1 : right2;
-        var br = !rightOrder ? right1 : right2;
-
-        goodFeaturesResultMat.release();
-        goodFeaturesResultMat.fromList(List.of(tl, bl, br, tr));
+    /**
+     * Refine an estimated corner position using the cornerSubPixel algorithm.
+     *
+     * TODO should this be here or before the points are chosen? 
+     *
+     * @param corners the corners detected -- this mat is modified!
+     * @param greyImg the image taken by the camera as color
+     * @return the updated mat, same as the corner mat passed in.
+     */
+    private MatOfPoint2f refineCornerEstimateSubPix(MatOfPoint2f corners, Mat greyImg) {
+        if(!shouldRefineCorners) return corners; // just return
+        Imgproc.cornerSubPix(greyImg, corners, winSize, zeroZone, criteria);
 
-        return goodFeaturesResultMat;
+        return corners;
     }
 
-    private StandardCVPipeline.TrackedTarget calculatePose(MatOfPoint2f imageCornerPoints, StandardCVPipeline.TrackedTarget target) {
+    NetworkTableEntry tvecE = NetworkTableInstance.getDefault().getTable("SmartDashboard").getEntry("tvec");
```

Comment:
Should this still be here? Maybe only enable it in debug mode? `--cv-development`

Reply:
commented it for now if its important @mcm001 can add it back